iOS/ViewControllers/Courses/PeerAssessmentViewController.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  Created for xikolo-ios under MIT license.
+//  Copyright © HPI. All rights reserved.
+//
+
+import Common
+import UIKit
+
+class PeerAssessmentViewController: UIViewController {
+
+    private static let pointsFormatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.decimalSeparator = "."
+        formatter.minimumFractionDigits = 1
+        formatter.maximumFractionDigits = 1
+        return formatter
+    }()
+
+    private static let dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "dd. MMM yyyy"
+        return formatter
+    }()
+
+    @IBOutlet private weak var peerAssessmentInfoView: UIStackView!
+    @IBOutlet private weak var deadlineMessageView: UIStackView!
+    @IBOutlet private weak var assessmentTitleLabel: UILabel!
+    @IBOutlet private weak var assessmentTypeLabel: UILabel!
+    @IBOutlet private weak var assessmentInstructionsView: UITextView!
+    @IBOutlet private weak var assessmentPointsLabel: UILabel!
+    @IBOutlet private weak var peerAssessmentTypeLabel: UILabel!
+    @IBOutlet private weak var peerAssessmentTypeImage: UIImageView!
+    @IBOutlet private weak var noteLabel: UILabel!
+    @IBOutlet private weak var redirectButton: UIButton!
+    @IBOutlet private weak var teamAssessmentView: UIStackView!
+    @IBOutlet private weak var soloAssessmentView: UIStackView!
+    @IBOutlet private weak var deadlineLabel: UILabel!
+    @IBOutlet private weak var deadlineDateView: UIStackView!
+
+    weak var delegate: CourseItemViewController?
+
+    private var courseItemObserver: ManagedObjectObserver?
+
+    private var courseItem: CourseItem! {
+        didSet {
+            self.courseItemObserver = ManagedObjectObserver(object: self.courseItem) { [weak self] type in
+                guard type == .update else { return }
+                DispatchQueue.main.async {
+                    self?.updateView()
+                }
+            }
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        var deadlineExpired = false
+
+        let formatter = DateFormatter()
+        formatter.dateFormat = "dd. MMM yyyy"
+
+        if let deadline = self.courseItem?.deadline {
+            deadlineExpired = deadline.inPast
+            self.deadlineDateView.isHidden = false
```

Comment:
I sort of dislike this approach. You rely on the initial value from the storyboard file. However, this value is hidden to the developer only looking at this file. Please make sure to also cover the opposite case (hiding the view), so the state is independent of the storyboard.

I would suggest something like this:
```swift
let deadlineExpired = self.courseItem?.deadline?.inPast ?? false
self.deadlineDateView.isHidden = self.courseItem?.deadline == nil
```